### ISSUE TITLE
New name suggestions?

### ISSUE THREAD

**Issue by renyuneyun** [renyuneyun/Easer#163]
So, there are some/many issues about the un-intuitive names for some elements of Easer. I also hate the bad names but don't really have good proposals.

Now, as I'm doing lots of refactoring to allow remote plugins in Easer (which will be presented in v0.7) and reading new users get confused, I also think it's time to get good names in this version (or soon).

#94 proposes some possible changes. However the names there include plurals which I don't think is a good idea.
From this proposal and some later thoughts, I have the following prepared, but expect everyone (interested) to spread your opinions.

Script ==> Task
Profile ==> Script

The motivation is:
* "Script" can often be seen as a series of actions, at least on computers. This lies exactly with what we currently call "Profile" intends to provide.
* However, if we rename Profile to Script, there is a naming conflict.
* The current "Script" provides two functions:
    1. A node in a state tree, containing the Event and Condition states
    2. An associate of actions to be performed with the node (when the node is "satisfied")
* (As also mentioned by @natrius ) "Task" can mean to "do something when something is met"

Any ideas?

p.s. discussion also welcomed in the [chatroom](https://matrix.to/#/#Easer:matrix.org) (see README)

**Comment by spaetz** [comment]
Hi there,
I would keep the profile name as is. profiles are used in Lineageos (and even possibly AOSP?) and people I talk to have understood a profile as a certain set of system settings to be used. which fits. 

What we call now script is perhaps better names 'trigger'? ie it describes the conditions and profiles and under which situation what action is triggered. I am a nonnative speaker myself though, so perhaps some English mothertongue could also weigh in here.

**Comment by natrius** [comment]
Not good imho. The trigger or event is what creates the task, action or a set of actions. Pull the trigger, weapon fires. Read NFC, Profile XY active. Time reached, profile YZ active.

**Comment by renyuneyun** [comment]
@natrius I think you meant to reply to @spaetz ?

@spaetz one of the ideas why I'd like to replace the name "Profile" is that "Profile" is considered "static" and only about "system settings" (at least from my point of view), but what this function intends to provide is a series of actions, which can include changing system settings, but can also include other actions (e.g. send network packet).
There is also a plan to allow perform/trigger/load "Profile B" in "Profile A", which seems to break the meaning of a LineageOS Profile.

By the way, there is a Remote Plugin (Operation) of Easer [which can load a LineageOS Profile](https://github.com/renyuneyun/EaserPlugin-LineageProfileSwitch). You can have a try if interested. It (/every Remote Plugin) acts as a normal action/Operation in a "Profile" in Easer, so you can combine it with other Operations too.

**Comment by natrius** [comment]
Yes, it was an reply to @spaetz , specially to the trigger part :)

**Comment by racuna** [comment]
I will change "Profile" to "Action" or "Task" (implying "do something"). Also, I will put "Events and Conditions" inside a section of "Triggers" (because is planned to get rid of the Events, IIRC)

Then I will have the main menu sorted this way:
- Triggers
- Actions
- Scripts (disabled until user creates some Trigger and an Action first)

**Comment by renyuneyun** [comment]
@racuna The plan is to remove some Events *which should better be expressed as Conditions*, not all Events ;p

Do you feel comfortable about the name "Script"? This is one of my main concerns about the names.

I prefer not to use plurals for the names, so "Action" doesn't seem like a good name; Condition doesn't seem like a *trigger* from my perspective. (I may be totally wrong about the meaning of "trigger".)

**Comment by renyuneyun** [comment]
I have a name emerged from #178 : We can move "Event" and "Condition" to one section called "Atom", to reflect the design purpose.

However the weak point is that "atom" do not reflect the utilization purpose...

**Comment by spaetz** [comment]
Actually, I like the term 'condition' a lot. 
Seriously, I think it is more important to stick to the terms than to rename them to something incrementally better. Easer is hard enough to understand and by renaming the core concepts all current users need to relearn things 😃. So a term change only if it is MUCH better than before,please...

**Comment by natrius** [comment]
Can't see how anyone would have an idea about "Atom" at first sight.
But i agree with apetz: Try to choose something you think is apropriate now and stick with it unless more and more people are complaining about a specific term.

And, i still don't see the problem with using plural. Es in every section multiple entrys are possible, it would be completely valid, using plural in english and german all over.
You said you don't have plural in your mother tongue - don't use plural there, if its not possible.

**Comment by wsdfhjxc** [comment]
I'll just say that I've also found the existing naming of things unintuitive. Now, I'm pretty sure my reply to this old thread won't have any influence, but I'm leaving some suggestions anyway. After all, the issue is still open, and maybe it can be revisited later.

Personally, I'd rename Script as Job, and Profile as Action. Then in the UI, instead of 4 tabs, I'd split things into 3 tabs:

| Tab                   | Description                                                                                 |
| --------------------- | ------------------------------------------------------------------------------------------- |
| Jobs                  | A Job starts an Action when a specified Condition is met, or when a specified Event occurs. |
| Actions               | An Action defines an operation, or a sequence of operations, which you want to be performed by the device (e.g. mute all sounds, send an SMS, launch an application, etc.). Actions are started by Jobs. |
| Events and Conditions | An Event represents a one-time occurrence (e.g. SMS received), and a Condition represents some state of the device (e.g. when battery level is below 15%). Events and Conditions are being monitored by Jobs. |

Yes, I'd put Events and Conditions together under a single tab, as they conceptually serve a similar purpose. They'd be distinctive in the list by some per-item labels or color markers, so that it's clear which one is an Event, and which one is a Condition.

That's all subjective of course, but as a first time user, I think I'd rather have things presented and described as in my suggestions above, than as it is currently. Nevertheless, the app is extremely useful, and thank you for developing it @renyuneyun.
